app/Http/Controllers/Bible/BibleVersesController.php

```diff
@@ -0,0 +1,164 @@
+     *          description="",
+     *     ),
+     *     @OA\Parameter(
+     *          name="book_id",
+     *          in="path",
+     *          @OA\Schema(ref="#/components/schemas/Book/properties/id"),
+     *          description="The book to filter bible_verses by",
+     *          example="MAT"
+     *     ),
+     *     @OA\Parameter(
+     *          name="chapter",
+     *          in="query",
+     *          @OA\Schema(ref="#/components/schemas/BibleVerse/properties/chapter"),
+     *          description="The chapter to filter bible_verses by"
+     *     ),
+     *     @OA\Parameter(
+     *          name="verse_number",
+     *          in="path",
+     *          @OA\Schema(ref="#/components/schemas/BibleVerse/properties/verse_number"),
+     *          description="The verse start to filter bible_verses by",
+     *          example="10"
+     *     ),
+     *     @OA\Parameter(ref="#/components/parameters/page"),
+     *     @OA\Parameter(ref="#/components/parameters/limit"),
+     *     @OA\Response(
+     *         response=200,
+     *         description="successful operation",
+     *         @OA\MediaType(mediaType="application/json", @OA\Schema(ref="#/components/schemas/v4_bible_verses.all"))
+     *     )
+     * )
+     * @return mixed
+     */
+    public function showVerseByLanguage(
+        string $language_code,
+        string $book_id,
+        string $chapter_id,
+        int $verse_number = null,
+    ) {
+        $limit          = (int) (checkParam('limit') ?? 15);
+        $limit          = min($limit, 50);
+        $page           = checkParam('page') ?? 1;
+        $key = $this->key;
+
+        $cache_params = [$limit, $page, $language_code, $key, $book_id, $chapter_id, $verse_number];
+        $cache_key = generateCacheSafeKey('bible_verses_by_language', $cache_params);
+        $verses = cacheRememberByKey(
+            $cache_key,
+            now()->addDay(),
+            function () use ($key, $limit, $language_code, $verse_number, $book_id, $chapter_id) {
+                return BibleVerse::withBibleFilesets($book_id, $chapter_id, $verse_number)
+                    ->filterByLanguage($language_code)
+                    ->isContentAvailable($key)
+                    ->paginate($limit);
+            }
+        );
+
+        return $this->reply(fractal(
+            $verses,
+            new BibleVerseTransformer(),
+        ));
+    }
+
+    /**
+     * Display a listing of the bibles.
+     *
+     * @OA\Get(
+     *     path="/bible/{bible_id}/verses/{book_id}/{chapter_id}/{verse_number?}",
+     *     tags={"BibleVerses"},
+     *     summary="Returns Bibles Verses based on filter criteria",
+     *     description="The base bible route returning by default bibles and filesets that your key has access to",
+     *     operationId="v4_bible_verses.verse_by_bible",
+     *     @OA\Parameter(
+     *          name="bible_id",
```

Comment:
book should be required

Reply:
It has been updated and it is a required parameter.

Reply:
@bradflood My bad, I missed adding more information, I have added the constraint to validate that some parameters should be required into the routing. In the following line you can see that I have used the methods: `whereAlphaNumeric` or `whereNumber` to validate the parameters through the routing. According to the above methods the parameters should pass the validations to go the controller.

https://github.com/faithcomesbyhearing/dbp/pull/838/files#diff-0e4323e6a03c91c9c8dc88afef15fdad09636468c0ac3550f497618372788b6dR130

```php
Route::name('v4_bible_verses.verse_by_language')->get(
    '/bibles/verses/{language_code}/{book_id}/{chapter_id}/{verse_number?}',
    'Bible\BibleVersesController@showVerseByLanguage'
)->whereAlphaNumeric('language_code')
->whereAlphaNumeric('book_id')
->whereNumber('chapter_id')
->whereNumber('verse_number');

Route::name('v4_bible_verses.verse_by_bible')->get(
    '/bible/{bible_id}/verses/{book_id}/{chapter_id}/{verse_number?}',
    'Bible\BibleVersesController@showVerseByBible'
)->whereAlphaNumeric('bible_id')
->whereAlphaNumeric('book_id')
->whereNumber('chapter_id')
->whereNumber('verse_number');
```